cmd/proxy/Dockerfile

```diff
@@ -1,15 +1,30 @@
-FROM golang:1.11
+FROM golang:1.11-alpine AS builder
 
-RUN mkdir -p $GOPATH/src/github.com/gomods/athens
 WORKDIR $GOPATH/src/github.com/gomods/athens
 
 COPY . .
 
-RUN CGO_ENABLED=0 GO111MODULE=on go build -mod=vendor -o /bin/app ./cmd/proxy
-RUN ./scripts/create_default_config.sh
+RUN apk update && \ 
+	apk add --no-cache bash && \
+	GO111MODULE=on CGO_ENABLED=0 go build -mod=vendor -o /bin/app ./cmd/proxy && \
+	./scripts/create_default_config.sh
+
+FROM alpine
+
+ENV GO111MODULE=on
+
+COPY --from=builder /bin/app /bin/app
+COPY --from=builder /go/src/github.com/gomods/athens/config.toml /config/config.toml
+COPY --from=builder /go/src/github.com/gomods/athens/cmd/proxy/locales /go/src/github.com/gomods/athens/cmd/proxy/locales
+COPY --from=builder /go/src/github.com/gomods/athens/cmd/proxy/assets /go/src/github.com/gomods/athens/cmd/proxy/assets
+COPY --from=builder /usr/local/go/bin/go /bin/go
+
+RUN apk update &&  \
+		apk add --no-cache bzr git mercurial openssh-client subversion procps && \
+		mkdir -p /usr/local/go
```

Comment:
why do we need this?

Reply:
this installs the VCS tools and GO is looking for it's default GOROOT which is /usr/local/go otherwise it fails to run.

Reply:
Copy go binary to /usr/local/go/bin/ and set PATH env ,
The compatibility of this will be better.